code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -365,35 +364,29 @@
 /mob/living/simple_animal/hostile/proc/DestroyStructuresCycle(turf/T)
 	var/dir_to_target = get_dir(targets_from, get_cardinal_step_towards(targets_from, target))
 	var/list/dir_list = list(dir_to_target, turn(dir_to_target, 90), turn(dir_to_target, 270), turn(dir_to_target, 180))
-	cycle_loop:
-		for(var/dir in dir_list)
-			T = get_step(targets_from, dir)
-			for(var/a in T)
-				var/atom/A = a
-				if(is_type_in_typecache(A, environment_target_typecache) && !A.IsObscured())
-					A.attack_animal(src)
-					break cycle_loop
+	for(var/dir in dir_list)
+		T = get_step(targets_from, dir)
+		for(var/a in T)
+			var/atom/A = a
+			if(is_type_in_typecache(A, environment_target_typecache) && !A.IsObscured())
+				A.attack_animal(src)
+				return
+
 
 /mob/living/simple_animal/hostile/proc/DestroyPathToTarget()
-	if(environment_smash)
-		EscapeConfinement()
```

Comment:
I said the if chain wasn't good, not that this wasn't.
this code is all that this should actually need to work.
````
	if(environment_smash)
		EscapeConfinement()
		DestroyStructuresCycle()
```

Reply:
yea i reverted it in my working code
i didnt understand what u meant by if chain

Reply:
It's a big chain of if() else statements that other code already handles(or should handle)